src/actions/chatActions.js

```diff
@@ -13,8 +13,13 @@ export const CHAT_USERNAME_SET = 'CHAT_USERNAME_SET';
 
 export const initChat = () => ({ type: CHAT_INIT_CHAT });
 
-export const initConnection = (pincode = '') =>
-  ({ type: CHAT_INIT_CONNECTION, pincode });
+export const initConnection = ({
+  authToken, secretKey, mID, isNewRoom
+}) =>
+  ({
+    type: CHAT_INIT_CONNECTION,
+    authToken, secretKey, mID, isNewRoom
+  });
```

Comment:
Step 5. This is the action dispatched by Step 4, when doing the auth. Now we enter the observable flow, when it receives this action type.